### PR TITLE
fix(pass): fix DCE in ExpandMixedKernel dropping AIV ops in mixed loops

### PR DESCRIPTION
**Pull request by Hzfengsy**
## Summary

- Fix four DCE blind spots in `ExpandMixedKernel` that caused live Vec operations to be incorrectly eliminated after AIC/AIV splitting in mixed loops with `init_values` (iter_args)
- `YieldStmt` now treated as a live root so yield-value computation chains (tpop + tile.add) are preserved
- `ForStmt`/`WhileStmt` `iter_args` init values now tracked for liveness so accumulator init definitions (tile.create + tile.muls) are preserved
- `tpop` ops now treated as side effects to prevent AIC/AIV queue desynchronization when tpop results are dead
- Add four regression tests covering single/multiple iter_args, bidirectional boundaries, and dead tpop results

## Root Cause

The `EliminateDeadCode` function inside `ExpandMixedKernel` had incomplete liveness analysis:

1. `FindLiveRootsRecursive` only checked `ReturnStmt` and side-effect ops — `YieldStmt` (a separate IR node) was ignored, so yield-value computation chains were eliminated
2. `ForStmt::iter_args_` and `WhileStmt::iter_args_` init value expressions were never walked for variable references, so init-value definitions were eliminated
3. `IsSideEffectOp` included `tpush` but not `tpop` — removing a `tpop` while keeping the corresponding `tpush` desynchronizes the AIC/AIV communication queues

These bugs affected all real-world models with GEMM accumulation loops (e.g., Qwen3 attention/MLP kernels).

## Testing

- [x] All 2321 existing tests pass
- [x] 4 new regression tests in `TestDCERegression` using Before/After `ir.assert_structural_equal`
- [x] Code review completed
- [x] Clang-tidy: 1 pre-existing warning (not in diff), no new issues

### CODE REVIEW

File: src/ir/transforms/expand_mixed_kernel_pass.cpp [hunk 1]
```diff
@@ -362,19 +363,34 @@ void CollectAllAssignStmts(const std::vector<StmtPtr>& stmts,
...
         FindLiveRootsRecursive(FlattenBody(if_stmt->else_body_.value()), live);
       }
     } else if (auto while_stmt = std::dynamic_pointer_cast<const WhileStmt>(stmt)) {
+      for (const auto& iter_arg : while_stmt->iter_args_) {
+        outline_utils::VarRefCollector refs;
+        refs.VisitExpr(iter_arg->initValue_);
+        live.insert(refs.var_refs.begin(), refs.var_refs.end());
+      }
       FindLiveRootsRecursive(FlattenBody(while_stmt->body_), live);
     }
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply src/ir/transforms/expand_mixed_kernel_pass.cpp]: Fixed in `4c6a421` — extracted `collect_iter_arg_refs` and `collect_expr_refs` lambdas to eliminate the duplication between ForStmt and WhileStmt.

File: tests/ut/ir/transforms/test_expand_mixed_kernel.py [hunk 2]
```diff
@@ -2507,5 +2478,400 @@ def main(
...
+
+
+class TestDCERegression:
+    """Regression tests for DCE bugs that dropped Vec ops in mixed loops.
+
+    These tests exercise the four DCE fixes:
+    1. YieldStmt treated as live root (yield-value computation chains preserved)
+    2. ForStmt iter_args init values tracked as live (init-value definitions preserved)
+    3. WhileStmt iter_args init values tracked as live (same as ForStmt)
+    4. tpop ops treated as side effects (AIC/AIV queue synchronization preserved)
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_expand_mixed_kernel.py:2493]: Acknowledged — the WhileStmt code path is symmetric with ForStmt (same lambda now). Adding a WhileStmt test requires complex DSL setup (`pl.while_` with `pl.cond`); deferring to a follow-up since the shared lambda ensures both paths stay in sync.

File: src/ir/transforms/expand_mixed_kernel_pass.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply src/ir/transforms/expand_mixed_kernel_pass.cpp]: Fixed in `4c6a421` — now rooting `start_/stop_/step_` for ForStmt, `condition_` for IfStmt/WhileStmt via `collect_expr_refs` lambda.

File: tests/ut/ir/transforms/test_expand_mixed_kernel.py [hunk 3]
```diff
@@ -2507,5 +2478,400 @@ def main(
...
+                w: pl.Tensor[[128, 128], pl.BF16],
+                out_0: pl.Out[pl.Tensor[[16, 128], pl.FP32]],
+            ) -> pl.Tensor[[16, 128], pl.FP32]:
+                acc_0: pl.Tile[[16, 128], pl.FP32, pl.MemorySpace.Vec, pl.TileView()] = pl.tile.create(
+                    [16, 128], dtype=pl.FP32, target_memory=pl.MemorySpace.Vec
+                )
+                acc_1: pl.Tile[[16, 128], pl.FP32, pl.MemorySpace.Vec, pl.TileView()] = pl.tile.muls(
+                    acc_0, 0.0
+                )
+                for i, (acc_iter,) in pl.range(4, init_values=(acc_1,)):
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_expand_mixed_kernel.py:2523]: Skipped — B007 is not enforced by the project's ruff config (CI passes), and renaming only in the new tests would be inconsistent with existing tests that also use `for i in pl.range(...)`.
